src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -112,6 +114,114 @@ class OPENASSETIO_CORE_EXPORT Manager final {
    * @{
    */
 
+  /**
+   * Capabilities that the manager implements.
+   *
+   * Many OpenAssetIO methods are optional. This enum is used with the
+   * introspection mechanism @ref hasCapability to provide a means of
+   * querying which sets of methods the manager provides.
+   */
+  enum class Capability : std::underlying_type_t<internal::capability::manager::Capability> {
+    /**
+     * Manager makes use of the context to persist custom state for
+     * performance reasons or otherwise.
+     *
+     * @note If this capability is true, then the host must reuse the
+     * same context across related API calls (including the use of @ref
+     * persistenceTokenForContext when the calls are distributed
+     * cross-process).
+     *
+     * @see @ref stable_resolution
+     */
+    kStatefulContexts = internal::capability::manager::Capability::kStatefulContexts,
+    /**
+     * Manager customizes certain human-readable strings that the host
+     * might want to use in UI/messages.
+     *
+     * This capability means the manager implements the following
+     * methods:
+     * - @ref updateTerminology
+     */
+    kCustomTerminology = internal::capability::manager::Capability::kCustomTerminology,
+    /**
+     * Manager is capable of resolving @ref entity_reference into the
+     * data for one or more @ref trait "traits".
+     *
+     * This capability means the manager implements the following
+     * methods:
+     * - @ref resolve
+     */
+    kResolution = internal::capability::manager::Capability::kResolution,
+    /**
+     * Manager allows the host to create or update an @ref entity within
+     * the @ref asset_management_system.
+     *
+     * This capability means the manager implements the following
+     * methods:
+     * - @ref preflight
+     * - @ref register_
+     */
+    kPublishing = internal::capability::manager::Capability::kPublishing,
+    /**
+     * Manager is capable of querying entity references that are related
+     * to the input references by the relationship defined by a set of
+     * traits and their properties.
+     *
+     * This capability means the manager implements the following
+     * methods:
+     * - @ref getWithRelationship
+     * - @ref getWithRelationships
+     * - @ref getWithRelationshipPaged
+     * - @ref getWithRelationshipsPaged
+     */
+    kRelationshipQueries = internal::capability::manager::Capability::kRelationshipQueries,
+    /**
+     * Manager is capable of confirming the existence of entities.
+     *
+     * This capability means the manager implements the following
+     * methods:
+     * - @ref entityExists
+     */
+    kExistenceQueries = internal::capability::manager::Capability::kExistenceQueries,
+
+    /**
+     * Manager may be capable of a providing an @ref EntityReference
```

Comment:
[Consider] Just "may" is too ambiguous here. The host needs advice on what that means specifically. I think the note here from before achieved that.

Reply:
Ah I think I found the text you're referring to

```
     * @note The fact that this capability is implemented does not
     * necessarily mean that a default entity reference can be provided
     * for every traitSet, you may get still get an empty value.
```

IIRC we removed this because it's covered by the method docstring itself, not to mention the method signature.

Perhaps something like "Manager is capable of responding appropriately to queries for an @ref EntityReference considered to be a sensible default"?

Reply:
Hmm, I guess that makes sense, although i'd personally favour over-specifying, even if it means repeating ourselves, especially as this type is becoming a sort of reference index.

Nonetheless though, the reasoning is valid, so resolving.